fix: Update dependencies, ensure build success and add axum-middleware-example

SiddheshKanawade opened the pull request:
Planned to ensure build with dependencies updated to latest version. Also add axum-middleware-example.

TODO:
- [x] Update dependencies to their latest version. solved #71
- [x] Add enable github actions on pull request.
- [x] Both tokio and async-std features of casbin are enabled, disable one to ensure casbin compile successfully
- [x] Make `actix-fileadapter-rbac` compile successfully after updating the dependencies
- [x] Remove the actix component version conflict and successfully compile `actix-middleware-example`
- [x] Remove the deprecated calls after updating dependencies and successfully compile `ntex-fileadapter-acl`
- [x] Add axum-middleware-example
- [x] Make clippy happy

solved #71

SiddheshKanawade committed:
upgraded dependencies in file adapter

SiddheshKanawade committed:
upgraded rest dependencies

hsluoyz commented:
@hackerchai

SiddheshKanawade commented:
@hackerchai do I need to do anything else?

hsluoyz commented:
@hackerchai plz review

hackerchai reviewed (approved):
LGTM

hackerchai commented:
@SiddheshKanawade Can you also add pull request github actions to run build checks?

SiddheshKanawade committed:
added pullrequest and cache in github actions

SiddheshKanawade commented:
@hackerchai I have added the pull_request check and also the cache, not sure the cache will work or not, last time it didn't worked

hsluoyz commented:
@SiddheshKanawade plz fix:

![image](https://user-images.githubusercontent.com/3787410/174481250-9daa4649-0f66-4fc7-8663-6dec6e2760e5.png)

SiddheshKanawade committed:
actix-fileadapter-rbac, actix-pgsql-simple, ntex-fileadapter-acl error resolved

SiddheshKanawade committed:
ntex-fileadapter-acl build success

SiddheshKanawade commented:
The above commits ensures that `actix-fileadapter-rbac, actix-pgsql-simple, ntex-fileadapter-acl` compiles successfully.

In actix-middleware-example, there are various version conflicts that I am trying to resolve(some examples below). Any suggestions on how should I resolve them? Also, the depedencies shown for actix-casbin-auth aren't updated in [crates.io,](https://crates.io/crates/actix-casbin-auth/0.4.4/dependencies) I think this might be the issue(not sure though). It says that v3 of actix-web is used and not v4.
```
actix-rt@1.1.1
actix-rt@2.7.0
```
```
actix-service@1.0.6
actix-service@2.0.2
```
```
actix-web@3.3.3
actix-web@4.1.0
```

SiddheshKanawade commented:
The possible cure here could be to match the versions of the dependencies to that of the crate.io but then it would restrict us to update the dependencies

hackerchai commented:
please fix

SiddheshKanawade commented:
> please fix

Yeah, I am working on it, its probably due to two different versions of tokio

SiddheshKanawade committed:
actix-middleware-example

SiddheshKanawade commented:
The crates.io don't have the versions of the dependencies updated, thus had to do following change to allow the use of newer versions of the actix dependencies without conflict
From
```toml
actix-casbin= {git = "https://github.com/casbin-rs/actix-casbin", default-features = false, features = [ "runtime-async-std" ]}
actix-casbin-auth = {git = "https://github.com/casbin-rs/actix-casbin-auth", default-features = false, features = [ "runtime-async-std" ]}
```
To
```toml
actix-casbin= {version = "0.4.2", default-features = false, features = [ "runtime-async-std" ]}
actix-casbin-auth = {version = "0.4.4", default-features = false, features = [ "runtime-async-std" ]}
```

This needs to be updated ones versions are updated in crates.io

SiddheshKanawade committed:
filestructure and initial db implementation

SiddheshKanawade committed:
corrected database test url

SiddheshKanawade committed:
utils, user model, user service

SiddheshKanawade committed:
tokio and async-std conflict resolved

SiddheshKanawade committed:
api, middleware, routes

SiddheshKanawade committed:
csv utils, main.rs, server setup

SiddheshKanawade committed:
register, update, delete services added

SiddheshKanawade committed:
make clippy happy, remove unnecessary commands

SiddheshKanawade commented:
@hackerchai I need to get PR [#25](https://github.com/casbin-rs/actix-casbin/pull/25) merged, so that we have updated tokio version in actix-casbin and avoid the tokio version conflict here. Also, I have completed my work on this PR and can be reviewed once the above mentioned PR gets merged.

hackerchai commented:
@SiddheshKanawade plz fix the CI errors

SiddheshKanawade committed:
renamed files

SiddheshKanawade committed:
Merge branch 'master' into master

SiddheshKanawade committed:
updated ci

SiddheshKanawade committed:
formatting, clippy happy in poem-todo

SiddheshKanawade commented:
![Screenshot from 2022-08-06 13-09-06](https://user-images.githubusercontent.com/85673539/183241571-00f96c97-880e-4b9a-a56e-961337b2e990.png)

I am not getting this error in my local env, any ideas on how should I cope with it?

SiddheshKanawade committed:
normalize::TrailingSlash->TrailingSlash

SiddheshKanawade commented:
@hackerchai All of the tasks have been completed and all checks are passing, kindly review and let me know if I need to do anything else

hsluoyz commented:
@SiddheshKanawade plz don't make so big and long-term PRs. You should not make just one PR for the whole summer and put everything in it. Our community's PRs are expected to be small-scale and will be reviewed & merged within 2-3 days, at most one week ideally.

SiddheshKanawade commented:
Yeah, actually this repository had a lot to do before I can make build run, so initially the PR became long. Also, this was what I was planning to do initially, but as per instructed by my mentor, I switched to axum-casbin-auth and hence this PR got extended. When I switched back to the task, rather than opening a new PR, I continued to work on this rather than closing it. 
Also, your suggestion makes a lot sense and will keep this in mind in future PRs.
Thank You.

hsluoyz commented:
OK.

@hackerchai plz review and get this one merged ideally

SiddheshKanawade commented:
Since now we have refactored this repository into smaller repositories, this PR has no significance and hence closing it.
